src/main/java/io/phdata/streamliner/schemadefiner/ConfigBuilder/ScriptCommand.java

```diff
@@ -0,0 +1,285 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ScriptCommand {
+  private static final Logger log = LoggerFactory.getLogger(ScriptCommand.class);
+
+  public static void build(
+      String configurationFile,
+      String configurationDiffFile,
+      String typeMappingFile,
+      String templateDirectory,
+      String outputDirectory)
+      throws IOException {
+
+    Configuration configuration = StreamlinerUtil.readConfigFromPath(configurationFile);
+    ConfigurationDiff configDiff = StreamlinerUtil.readConfigDiffFromPath(configurationDiffFile);
+    Map<String, Map<String, String>> typeMapping =
+        StreamlinerUtil.readTypeMappingFile(typeMappingFile);
+    if (outputDirectory == null || outputDirectory.equals("")) {
+      outputDirectory =
+          "output/" + configuration.getName() + "/" + configuration.getEnvironment() + "/scripts";
+    }
+    build(configuration, configDiff, typeMapping, templateDirectory, outputDirectory);
+  }
+
+  private static void build(
+      Configuration configuration,
+      ConfigurationDiff configDiff,
+      Map<String, Map<String, String>> typeMapping,
+      String templateDirectory,
+      String outputDirectory)
+      throws IOException {
+    if (configuration == null && configDiff == null) {
+      throw new RuntimeException(
+          "Check the configuration (example: streamliner-configuration.yml) file. Tables section is not found. Or Check the configuration difference (example: streamliner-configuration-diff.yml) file. TableDiff section is not found.");
+    }
+    String pipeline =
+        configuration == null ? configDiff.getPipeline() : configuration.getPipeline();
+    List<File> files = StreamlinerUtil.listFilesInDir(templateDirectory + "/" + pipeline);
+    List<File> templateFiles =
+        files.stream().filter(f -> f.getName().endsWith(".ssp")).collect(Collectors.toList());
+    List<File> nonTemplateFiles =
+        files.stream().filter(f -> !f.getName().endsWith(".ssp")).collect(Collectors.toList());
+
+    if (configuration != null && configuration.getTables() != null) {
+      log.info(
+          "Configuration (example: streamliner-configuration.yml) file found. Processing Table section...");
+      configuration.getTables().stream()
+          .forEach(
+              table -> {
+                String tableDir = outputDirectory + "/" + table.getDestinationName();
+                StreamlinerUtil.createDir(tableDir);
+
+                templateFiles.stream()
+                    .forEach(
+                        templateFile -> {
+                          log.info("Rendering file: {} ", templateFile);
+                          String rendered = "";
+                          if (table instanceof SnowflakeTable) {
+                            SnowflakeTable snowflakeTable = (SnowflakeTable) table;
```

Comment:
Is this map necessary? The casted table is going right back to an `object`

Reply:
this map is used to store all the attributes that template needs and casted table snowflakeTable is one of the attribute. Since the value could be of any type I have kept it as Object. But in template I am getting SnowflakeTable.

Reply:
Looks like it doesn't matter what type of object table is though. I think we can just case to TableDefinition and not use the concrete types.

Reply:
I think here we should use concrete types because HadoopTable and SnowflakeTable have their specific methods like columnDDL or there are few others.

Reply:
Yeah I think if we are using any of the methods specific to the type, then what we are doing here is exactly what we want to be doing. However, when I look at the code, I don't see anything specific to the type? Could you point out what work here is specific to the concrete type?

Reply:
HadoopTable : https://github.com/phdata/streamliner/blob/master/src/main/scala/io/phdata/streamliner/configuration/package.scala#L172
SnowflakeTable : https://github.com/phdata/streamliner/blob/master/src/main/scala/io/phdata/streamliner/configuration/package.scala#L225
Here we can see specific methods

Reply:
Anish and I discussed. He's going to resolve this.